src/coordinator/storage/local/storage.go

```diff
@@ -90,9 +143,54 @@ func (s *localStorage) FetchTags(ctx context.Context, query *storage.FetchQuery,
 		return nil, err
 	}
 
-	opts := storage.FetchOptionsToM3Options(options, query)
+	var (
+		opts       = storage.FetchOptionsToM3Options(options, query)
+		namespaces = s.clusters.ClusterNamespaces()
+		now        = time.Now()
+		fetches    = 0
+		result     multiFetchTagsResult
+		wg         sync.WaitGroup
+	)
+	for _, namespace := range namespaces {
+		namespace := namespace // Capture var
+
+		clusterStart := now.Add(-1 * namespace.Attributes().Retention)
```

Comment:
instead of this, can you just check if !clusterStart.Before(query.Start) {...}

Reply:
I want to include the case where query.Start matches the clusterStart however, which that logic would not cover.